indico/web/client/js/react/components/PersonLinkField.jsx

```diff
@@ -0,0 +1,400 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2021 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import PropTypes from 'prop-types';
+import React, {useMemo, useState} from 'react';
+import {Button, Segment, List, Form, Label, Icon, Popup, Message} from 'semantic-ui-react';
+
+import {UserSearch} from 'indico/react/components/principals/Search';
+import {PrincipalType} from 'indico/react/components/principals/util';
+import {FinalDropdown, FinalInput, FinalTextArea} from 'indico/react/forms';
+import {FinalModalForm} from 'indico/react/forms/final-form';
+import {useFavoriteUsers} from 'indico/react/hooks';
+import {snakifyKeys} from 'indico/utils/case';
+
+import {Translate} from '../i18n';
+
+import {PrincipalItem} from './principals/items';
+
+import './PersonLinkField.module.scss';
+
+const titles = [
+  {text: Translate.string('None'), value: ''},
+  {text: Translate.string('Mr'), value: 'Mr'},
+  {text: Translate.string('Ms'), value: 'Ms'},
+  {text: Translate.string('Mrs'), value: 'Mrs'},
+  {text: Translate.string('Dr'), value: 'Dr'},
+  {text: Translate.string('Prof'), value: 'Prof'},
+  {text: Translate.string('Mx'), value: 'Mx'},
+];
+
+const ExternalPersonModal = ({onSubmit, onClose, person}) => (
+  <FinalModalForm
+    id="person-link-details"
+    size="tiny"
+    onClose={onClose}
+    onSubmit={onSubmit}
+    header={Translate.string('Enter Person')}
+    initialValues={person || {}}
+  >
+    {person && person.userId && (
```

Comment:
indico instances migrated from 1.x have a user with ID `0`, so this needs to be an explicit `null` check

Reply:
Good point!